spec/quartz/config_spec.cr

```diff
@@ -0,0 +1,65 @@
+require "../spec_helper"
+
+module Quartz
+  def self.reset_config
+    @@config = Config.new
+  end
+end
+
+describe Quartz::Config do
+  context "smtp connection parameters" do
+    it "defaults to localhost" do
+      Quartz.reset_config
+      Quartz.config.smtp_address.should eq "127.0.0.1"
+    end
+
+    it "defaults to port 25" do
+      Quartz.reset_config
+      Quartz.config.smtp_port.should eq 25
+    end
+
+    it "defaults to ssl disabled" do
+      Quartz.reset_config
+      Quartz.config.use_ssl.should be_false
+    end
+  end
+
+  context "logger" do
+    it "builds a default logger" do
+      Quartz.reset_config
+      Quartz.config.logger.should be_a Logger
+    end
+  end
+
+  context "enable/disable" do
+    it "defaults to disabled" do
+      Quartz.reset_config
+      Quartz.config.smtp_enabled.should be_false
+    end
+  end
+
+  context "setting config parameters" do
+    it "allows config settings to be changed" do
+      alternate_reality = "127.1.1.1"
+      mailcatcher_port = 1080
+      happy_logger = Logger.new nil
+
+      Quartz.reset_config
+
+      Quartz.config.smtp_address = alternate_reality
+      Quartz.config.smtp_address.should eq alternate_reality
+
+      Quartz.config.smtp_port = mailcatcher_port
+      Quartz.config.smtp_port.should eq mailcatcher_port
+
+      Quartz.config.use_ssl = true
+      Quartz.config.use_ssl.should eq true
+
+      Quartz.config.smtp_enabled = true
+      Quartz.config.smtp_enabled.should be_true
+
+      Quartz.config.logger = happy_logger
+      Quartz.config.logger.should eq happy_logger
```

Comment:
It looks like concurrency is built-in.  We should probably add these settings:
```
Quartz.config.number_of_connections
Quartz.config.messages_per_connection
```

Reply:
As it stands, the connection doesn't get persisted past the sending of one email. However, the email is sent in a fiber! I know very little about Crystal fibers, but I think that means that you don't _need_ to have a job runner to do it outside of the request.

The concurrency of EMail means the Mailer class could very well be built to sustain a set of connections and send emails over those connections, but that's a big enough change to the flow that I think it might be best addressed in a future upgrade.